.github/workflows/test.yml

```diff
@@ -0,0 +1,170 @@
+            # Use chartpress to create the helm chart and build its images
+            helm dependency update ./helm-chart/binderhub
+            (cd helm-chart && chartpress)
+            git --no-pager diff
+      - name: Validate the chart against the k8s API
+        if: matrix.test == 'helm'
+        run: |
+            helm template --validate binderhub-test helm-chart/binderhub \
+                --values testing/k8s-binder-k8s-hub/binderhub-chart-config.yaml \
+                --set config.BinderHub.hub_url=http://localhost:30902 \
+                --set config.BinderHub.access_token=$GITHUB_ACCESS_TOKEN
+      - name: Install the chart
+        if: matrix.test == 'helm'
+        run: |
+          helm upgrade --install binderhub-test helm-chart/binderhub \
+              --values testing/k8s-binder-k8s-hub/binderhub-chart-config.yaml \
+              --set config.BinderHub.hub_url=http://localhost:30902 \
+              --set config.BinderHub.hub_url_local=http://proxy-public \
+              --set config.BinderHub.access_token=$GITHUB_ACCESS_TOKEN
+      - name: Wait for JupyterHub to be ready
+        run: |
+            # Wait for JupyterHub to be ready
+            set -e
+            . ci/common
+            await_jupyterhub
+            # Next few commands are here for sanity checking/debugging of CI
+            # related issues
+            echo "Proxy service details"
+            kubectl get svc proxy-public -o json
+            echo
+            echo "kubectl describe svc proxy-public"
+            kubectl describe svc proxy-public
+            echo
+            echo "kubectl get svc"
+            kubectl get svc
+            echo
+            echo "curl http://localhost:30902/hub/api/" should print the JupyterHub version
+            curl http://localhost:30902/hub/api/
+      - name: Wait for binderhub to be ready
+        if: matrix.test == 'helm'
+        run: |
+            # Wait for BinderHub to be ready
+            set -e
+            . ci/common
+            await_binderhub
+            echo "curl http://localhost:30901/health" to check BinderHub\'s health
+            curl http://localhost:30901/health
+            echo
+      - name: Run main tests
+        if: matrix.test == 'main'
+        # running the "main" tests means "all tests that aren't auth"
+        run: pytest -m "not auth" -vx --cov binderhub
+      - name: Run auth tests
+        if: matrix.test == 'auth'
+        # running the "auth" tests means "all tests that are marked as auth"
+        run: pytest -m "auth" -vx --cov binderhub
+      - name: Run helm tests
+        if: matrix.test == 'helm'
+        run: |
+            export BINDER_URL=http://localhost:30901
+            pytest -m "remote" -vx --cov binderhub
+      - name: Kubernetes namespace report
+        if: ${{ failure() }}
+        run: |
+          # Display debugging information
+          . ci/common
+          full_namespace_report
+      - name: "Upload code coverage stats"
```

Comment:
There's a [codecov action](https://github.com/codecov/codecov-action) that I've successfully used in
https://github.com/manics/jupyterhub-ansiblespawner/blob/4efa38c2016501675dabd23a269562d4af7ca1a1/.github/workflows/workflow.yml#L64-L67
Could be an option? But in the interests of fixing CI it's also fine to leave this for a later.

Reply:
I used this explicit snippet because I had it handy from trying to figure out why codecov (the action? the tool?) wasn't working for https://github.com/jupyterhub/repo2docker/pull/982

Switched to the action here